cli/main.rs

```diff
@@ -178,114 +181,273 @@ fn print_cache_info(state: &GlobalState, json: bool) -> Result<(), ErrBox> {
   }
 }
 
-struct FileInfoOutput<'a> {
-  local: &'a str,
-  file_type: &'a str,
-  compiled: Option<String>,
-  map: Option<String>,
-  deps: Option<Deps>,
+/// A dependency tree of the basic module information.
+///
+/// Constructed from a `ModuleGraph` and `ModuleSpecifier` that
+/// acts as the root of the tree.
+struct FileInfoDepTree {
```

Comment:
`#[derive(Serialize)]`?

Reply:
This was the original implementation. It was pointed out that there is a fair amount of redundancies in the JSON output that was removed in the console output. So I opted to remove these in a custom implementation of the serialize trait.

Alternatively, assuming the JSON is to be consumed by something else, it was suggested to just output the adjacency list from the graph with node metadata containing sizes and letting consumers choose how they want to use that size data.

Here is the comment:
https://github.com/denoland/deno/pull/6786#issuecomment-663913828_

Reply:
JSON output is not yet set in stone, so I think using `Serialize` is preferable to adding a custom implementation (of course if it handles circular dependencies properly).

Reply:
Sounds good.